src/main/java/gg/uhc/uhc/ShutdownListenerRegistry.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Project: UHC
+ * Class: gg.uhc.uhc.inventory.IconInventory
```

Comment:
Nice class name :P I'd just make this a static hook so we can run ShutdownListener.registerListener() from any class and in onDisable we can call a package private shutdown() method that will trigger all listeners and clear the list